.github/workflows/apply-terraform.yml

```diff
@@ -59,13 +64,6 @@ jobs:
       with:
         path: main
 
-    - name: Configure AWS credentials
```

Comment:
I'm struggling to get my head around why we don't need to auth here again (like is happening for the docker build step). Wouldn't this just use the EC2 instance profile role without running a further step to re-authenticate with OIDC credentials?

I know we do this in the `start-runner` phase, but that only kicks off the process to start the actual runner environment. I might be missing something (this is highly likely) but it looks like the runner env doesn't have any context around the OIDC role at this point? Let me know if I'm misinterpreting, I'm keen to fully understand the flow

Reply:
Thanks for spotting this. This was actually me testing to see where or not it was inheriting the credentials between the stages hence the `run: aws sts get-caller-identity` further down the script, as I wanted to reduce the amount of calls we had to make it. 

I need to readd that auth step here, I was also worried about speed, but the assume role happens within seconds of even being able to load the job up!